src/main/java/com/synopsys/integration/detect/ApplicationUpdater.java

```diff
@@ -284,6 +307,35 @@ private void checkEnvironmentProperties() {
             proxyPropertiesPut(ARG_PROXY_USERNAME, utility.getSysEnvProperty(SYS_ENV_PROP_PROXY_HTTP_USERNAME));
             proxyPropertiesPut(ARG_PROXY_PASSWORD, utility.getSysEnvProperty(SYS_ENV_PROP_PROXY_HTTP_PASSWORD));
         }
+        
+        if (!springbootEnvMap.isEmpty()) {
+            updateFromSpringBootPropertiesIfAny();
+        }
+    }
+    
+    private void updateFromSpringBootPropertiesIfAny() {
+        if (springbootEnvMap.containsKey(ARG_BLACKDUCK_URL)) {
+            blackduckHost = springbootEnvMap.get(ARG_BLACKDUCK_URL);
+        }
+        if (springbootEnvMap.containsKey(ARG_BLACKDUCK_OFFLINE_MODE)) {
+            offlineMode = springbootEnvMap.get(ARG_BLACKDUCK_OFFLINE_MODE);
+        }
+        if (springbootEnvMap.containsKey(ARG_TRUST_CERTIFICATE)) {
+            trustCertificate = Boolean.parseBoolean(springbootEnvMap.get(ARG_TRUST_CERTIFICATE));
+        }
+        updateProxyPropertyIfSpringKeyExists(ARG_PROXY_HOST);
+        updateProxyPropertyIfSpringKeyExists(ARG_PROXY_IGNORED_HOSTS);
+        updateProxyPropertyIfSpringKeyExists(ARG_PROXY_NTLM_DOMAIN);
+        updateProxyPropertyIfSpringKeyExists(ARG_PROXY_NTLM_WORKSTATION);
+        updateProxyPropertyIfSpringKeyExists(ARG_PROXY_PASSWORD);
+        updateProxyPropertyIfSpringKeyExists(ARG_PROXY_PORT);
+        updateProxyPropertyIfSpringKeyExists(ARG_PROXY_USERNAME);
+    }
+    
+    private void updateProxyPropertyIfSpringKeyExists(String key) {
+        if (springbootEnvMap.containsKey(key)) {
+            proxyPropertiesPut(key, springbootEnvMap.get(key));
+        }
```

Comment:
Method to update the Proxy Properties Map if same property is available in the Spring Boot Application Map.